src/object/obj_tx.c

```diff
@@ -3214,17 +3215,34 @@ dc_tx_convert(struct dc_object *obj, enum obj_rpc_opc opc, tse_task_t *task)
 	args->th = dc_tx_ptr2hdl(tx);
 	args->flags = 0;
 
+	/*
+	 * Make the @task to depend on the @tx_task before dc_tx_attach(), that will prevent
+	 * the @task being released after dc_tx_attach() returned because of the dependency.
+	 */
 	rc = dc_task_depend(task, 1, &tx_task);
 	if (rc != 0) {
 		D_ERROR("Fail to add dep on TX convert task: "DF_RC"\n",
 			DP_RC(rc));
 		goto out;
 	}
 
+	dep = true;
+
+	rc = dc_tx_attach(dc_tx_ptr2hdl(tx), obj, opc, task);
```

Comment:
[Question] How are we making sure that any existence check tasks are completed before starting the commit task?

Reply:
Good question, I will give more consideration for the patch.